File: fix_misc.js
const fs = require('fs');

let js = fs.readFileSync('js/platform.js', 'utf-8');

js = js.replace("GRK.confirm('Deseja sair da sua conta?', async () => {", "if (confirm('Deseja sair da sua conta?')) {");

const oldPush = `    document.getElementById('pushToggle')?.addEventListener('click', async () => {
        if (window.PWA) {
            try {
                await PWA.solicitarPermissaoPush();
                GRK.toast('Notificações ativadas! 🔔', 'success');
                carregarConfig();
            } catch (e) {
                GRK.toast('Permissão negada', 'error');
            }
        }
    });`;

const oldPush2 = `    document.getElementById('pushToggle')?.addEventListener('click', async () => {
        if (window.PWA) {
            try {
                await PWA.solicitarPermissaoPush();
                GRK.toast('Notificações ativadas! 🔔', 'success');
                carregarConfig();
            } catch (e) {
                GRK.toast('Permissão negada', 'error');
            }
        }
    });`.replace('🔔', '??'); // sometimes encoding issues

// We'll just do a more robust replace for pushToggle
js = js.replace(/document\.getElementById\('pushToggle'\)\?\.addEventListener\('click', async \(\) => \{[\s\S]*?\}\);/, `document.getElementById('pushToggle')?.addEventListener('click', async () => {
        if ('Notification' in window && Notification.permission === 'granted') {
            return GRK.toast('Já está ativo! Para desativar, mude nas permissões do navegador.', 'info');
        }
        if (window.PWA) {
            try {
                await PWA.solicitarPermissaoPush();
                GRK.toast('Notificações ativadas! 🔔', 'success');
                carregarConfig();
            } catch (e) {
                GRK.toast('Permissão negada', 'error');
            }
        }
    });`);

fs.writeFileSync('js/platform.js', js, 'utf-8');
console.log("Fixed fazerLogout and pushToggle");

File: fix_misc.py
import sys

with open('js/platform.js', 'r', encoding='utf-8') as f:
    js = f.read()

# Fix fazerLogout
js = js.replace('GRK.confirm(', 'if(confirm(')
# We need to make sure the if block is closed properly if it was a callback
# wait, the original was:
# GRK.confirm('Deseja sair da sua conta?', async () => {
#    ...
# });
# We should just replace GRK.confirm with confirm and run the code inline

js = js.replace("GRK.confirm('Deseja sair da sua conta?', async () => {", "if (confirm('Deseja sair da sua conta?')) {")

# Fix pushToggle
old_push = """    document.getElementById('pushToggle')?.addEventListener('click', async () => {
        if (window.PWA) {
            try {
                await PWA.solicitarPermissaoPush();
                GRK.toast('Notifica\u00e7\u00f5es ativadas! \ud83d\udd14', 'success');
                carregarConfig();
            } catch (e) {
                GRK.toast('Permiss\u00e3o negada', 'error');
            }
        }
    });"""

new_push = """    document.getElementById('pushToggle')?.addEventListener('click', async () => {
        if ('Notification' in window && Notification.permission === 'granted') {
            return GRK.toast('Ja esta ativo! Para desativar, mude nas permissoes do site.', 'info');
        }
        if (window.PWA) {
            try {
                await PWA.solicitarPermissaoPush();
                GRK.toast('Notificacoes ativadas! 🔔', 'success');
                carregarConfig();
            } catch (e) {
                GRK.toast('Permissao negada', 'error');
            }
        }
    });"""

js = js.replace(old_push, new_push)

with open('js/platform.js', 'w', encoding='utf-8') as f:
    f.write(js)

print("Fixed fazerLogout and pushToggle")
